airbyte-integrations/connectors/source-file/source_file/client.py

```diff
@@ -321,11 +323,16 @@ def load_dataframes(self, fp, skip_data=False) -> Iterable:
         reader_options = {**self._reader_options}
         try:
             if self._reader_format == "csv":
+                bytes_read = 0
                 reader_options["chunksize"] = self.CSV_CHUNK_SIZE
                 if skip_data:
                     reader_options["nrows"] = 0
                     reader_options["index_col"] = 0
-                yield from reader(fp, **reader_options)
```

Comment:
Read only `self.CSV_CHUNK_SIZE` bytes of data to generate schema. Otherwise a time out is possible in case a large file is read